client/services/userService.js

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable consistent-return */
+const axios = require('axios')
+
+/**
+ * @description The api endpoint for the user service request. It is set in the
+ * dot env file and imported from there with the given name.
+ * @constant baseUrl
+ * @type string
+ */
+const baseUrl = process.env.REACT_APP_BASE_USER_URL
+
+/**
+ * @name addFriend
+ * @summary The addFriend function adds a new friend to a user's friends array. The
+ * function accepts three parameters---user, id, and token. The request is sent to
+ * the backend, the user's friends list is updated and the updated list is returned.
+ * @param {{
+ * username: string,
+ * email: string,
+ * friends: string[],
+ * picture: Buffer
+ * }} user This is the user currently logged in
+ * @param {string} id This is the id of the current user
+ * @param {string} token This is token to authenticate the user before handling request
+ * @returns The updated list of the current user's friends
+ */
+const addFriend = async (user, id, token) => {
+  try {
+    // make a post request to the backend
+    const response = await axios.post(baseUrl, { ...user, id, token })
```

Comment:
- POST requests are reserved to creating users, update operations must send a PUT request.
- All update requests send the entire user object, for the reasons we spoke about in the meeting. It requires less communication with the backend. If Irene is following the specifications, the put request simply replaces the user in db with the received user, and returns the updated user.
- addFriend and unFriend are update operations, and they must send a PUT request.
- The id is a parameter and must be included in the url
- The token is sent as a config object
- The put request is sent in the following manner:

```
const config = {
    headers: { Authorization: `bearer ${ token }` },
  }
await axios.put(`${baseUserUrl}/id`, user, config)
```

Reply:
Log in request are usually `post` request even though no resources are created. Whenever you want to send a payload along with your request, post can do that. However, `PUT` will fit the scenario you mentioned in the meeting because it is syntactically the correct HTTP VERB for what the request is doing. Thanks.

Reply:
No problem. I believe Irene created a POST routr in the user router that creates users. She said she will add a PUT route to update users, so we must implement our update operations using PUT.